modules/actions/snapshots/sync-snapshots.ts

```diff
@@ -0,0 +1,119 @@
+import { Chain, Prisma, PrismaLastBlockSyncedCategory } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import _ from 'lodash';
+import { PoolSnapshot_Filter } from '../../sources/subgraphs/balancer-v3-vault/generated/types';
+import { fillMissingTimestamps } from './lib/fill-missing-timestamps';
+import { computeDailyValues } from './lib/compute-daily-values';
+import { applyUSDValues } from './lib/apply-usd-values';
+
+type SnapshotsSubgraphClient = {
+    getMetadata: () => Promise<{ block: { number: number } }>;
+    getAllSnapshots: (args: PoolSnapshot_Filter) => Promise<Prisma.PrismaPoolSnapshotUncheckedCreateInput[]>;
+};
+
+export async function syncSnapshots(
+    subgraphClient: SnapshotsSubgraphClient,
+    category: PrismaLastBlockSyncedCategory,
+    chain: Chain,
+    options: {
+        poolIds?: string[]; // Optional: Specific pool IDs to sync
+        syncLatest?: boolean; // Optional: Whether to sync based on the latest block
+    } = {},
+): Promise<string[]> {
+    const { poolIds = [], syncLatest = true } = options;
+    const currentBlock = await subgraphClient.getMetadata().then((meta) => meta.block.number);
+
+    let lastBlock = 0;
+    if (syncLatest) {
+        lastBlock = await prisma.prismaLastBlockSynced
+            .findFirst({
+                where: { chain, category },
+                select: { blockNumber: true },
+            })
+            .then((lastBlock) => lastBlock?.blockNumber || 0);
+    }
+
+    // Fetch pool IDs from DB
+    const poolIdsInDb = await prisma.prismaPool
+        .findMany({ where: { chain } })
+        .then((pools) => pools.map((pool) => pool.id));
+
+    // Fetch snapshots from subgraph client
+    const snapshots = await subgraphClient
+        .getAllSnapshots({
+            pool_: { isInitialized: true },
+            ...(syncLatest && lastBlock > 0 ? { _change_block: { number_gte: lastBlock } } : {}),
+            ...(poolIds.length > 0 ? { pool_in: poolIds } : {}),
```

Comment:
this might explode if we have a lot of pool ids

Reply:
i know, but i'll take care of it later

Reply:
it's an optional param anyways, so actually it's on purpose. by default it's not passing poolIds filter